algebird-core/src/main/scala/com/twitter/algebird/CuckooFilter.scala

```diff
@@ -0,0 +1,435 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.twitter.algebird
+
+import algebra.BoundedSemilattice
+import com.googlecode.javaewah.{IntIterator, EWAHCompressedBitmap => CBitSet}
+
+import scala.util.Random
+
+/**
+ * Cuckoo filter for Algebird in a Monoid way of life.
+ * referring :
+ *  - https://github.com/twitter/algebird/issues/560#issue-185161455
+ *  - http://www.eecs.harvard.edu/%7Emichaelm/postscripts/cuckoo-conext2014.pdf
+ *  - https://github.com/irfansharif/cfilter
+ * By nature, this filter isn't commutative
+ * From the inital paper, there is no problem to consider || fingerprint|| = ln(N) where N = fingerprintPerBucket * totalBucket
+ * "" as long as we use reasonably sized buckets, the fingerprint size can remain small. "", we'll use a a 32 bits fingerprint.
```

Comment:
Original paper suggests that fingerprint is best below 8 bits. Any reason you're using 32?